management/executive/src/executive.cc

```diff
@@ -3348,6 +3434,82 @@ bool Executive::Undock(ff_msgs::CommandStampedPtr const& cmd) {
   return docked;
 }
 
+bool Executive::UnloadNodelet(ff_msgs::CommandStampedPtr const& cmd) {
+  NODELET_INFO("Executive executing unload nodelet command!");
+  bool success = true;
+  std::string err_msg = "Unload nodelet parameters must be strings.";
+  ff_msgs::UnloadLoadNodelet unload_load_nodelet_srv;
+  unload_load_nodelet_srv.request.load = false;
+
+  // Don't unload a nodelet while moving because we may be unloading a mobility
+  // nodelet
+  if (CheckNotMoving(cmd)) {
```

Comment:
What happens if we are moving? It doesn't look like it gives an error with an error message?

Reply:
The check not moving function sends a failed ack i.e. it does give an error message.

Reply:
Is that the right behavior? Seems like it could be confusing.

Reply:
This function is only used to check if the robot is not moving in terms of executing a command. It seemed cleaner to ack in the function rather than having the same ack if chain in multiple command functions. Would it be clearer if the name was changed to CheckNotMovingAndFailAckIfMoving?

Reply:
Renamed CheckNotMoving to FailCommandIfMoving.